agenda/tests.py

```diff
@@ -16,52 +16,50 @@
 """
 from gig.tests import GigTestBase
 from gig.util import GigStatusChoices, PlanStatusChoices
+from gig.models import Plan
+from member.util import AgendaLayoutChoices
+from agenda.helpers import _get_agenda_plans
+from band.models import Band, Assoc
+from band.util import AssocStatusChoices
 from django.test import Client
 from django.urls import reverse
 from json import loads
 from datetime import datetime
 from pytz import timezone
 
 class AgendaTest(GigTestBase):
-    def test_agenda(self):
+    def test_agenda_types(self):
+        """ Test the different URLs for returning panels on the agenda page """
         self.assoc_user(self.joeuser)
         gigs = []
         for i in range(0, 19):
             gigs.append(self.create_gig_form(contact=self.joeuser, title=f"xyzzy{i}"))
         c = Client()
         c.force_login(self.joeuser)
+        self.joeuser.preferences.agenda_layout = AgendaLayoutChoices.ONE_LIST
+        self.joeuser.preferences.save()
 
-        # first 'page' of gigs should have 19 (because pagination is very long)
-        response = c.get(f'/plans/noplans/1')
+        # get as a single list
+        response = c.get(f'/plans/{int(AgendaLayoutChoices.ONE_LIST)}/0')
         self.assertEqual(response.content.decode('ascii').count("xyzzy"), 19)
 
+        # test "has plans" vs. "doesn't have plans"
+
         first_gig = gigs[0]
         plan = first_gig.plans.get(assoc__member=self.joeuser)
         plan.status = PlanStatusChoices.DEFINITELY
         plan.save()
-        response = c.get(f'/plans/noplans/1')
+
+        response = c.get(f'/plans/{int(AgendaLayoutChoices.NEED_RESPONSE)}/0')
         self.assertEqual(response.content.decode('ascii').count("xyzzy"), 18)
-        response = c.get(f'/plans/plans/1')
-        self.assertEqual(response.content.decode('ascii').count("xyzzy"), 1)
 
         # now make sure that a gig that is canceled is moved to the "plans" set even
         # though it has no plans...
         gigs[1].status = GigStatusChoices.CANCELED
         gigs[1].save()
-        response = c.get(f'/plans/noplans/1')
+        response = c.get(f'/plans/{int(AgendaLayoutChoices.NEED_RESPONSE)}/0')
         self.assertEqual(response.content.decode('ascii').count("xyzzy"), 17)
-        response = c.get(f'/plans/plans/1')
-        self.assertEqual(response.content.decode('ascii').count("xyzzy"), 2)
-
-
-        # tests that pass if pagination is set to 10        
-        # # first 'page' of gigs should have 10
-        # response = c.get(f'/plans/noplans/1')
-        # self.assertEqual(response.content.decode('ascii').count("xyzzy"), 10)
 
-        # # second 'page' of gigs should have 9
-        # response = c.get(f'/plans/noplans/2')
-        # self.assertEqual(response.content.decode('ascii').count("xyzzy"), 9)
```

Comment:
Should these tests be fixed or removed?

Reply:
They're removed already - not sure why you see them.